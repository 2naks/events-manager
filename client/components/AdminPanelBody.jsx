import React, { Component } from 'react';
import AdminCenters from './AdminCenters.jsx';
import AdminCenterDetails from './AdminCenterDetails.jsx';
import AddCenter from './AddCenter.jsx';
import ModifyCenter from './ModifyCenter.jsx';

class AdminPanelBody extends Component {

  constructor(props) {
    super(props);
    this.state = {
      centers: {
        visible: true
      },
      addCenter: {
        visible: false
      },
      modifyCenter: {
        visible: false
      },
      centerDetails: {
        visible: false
      }
    };
  }

  show = (elem) => {
    let hiddenElems = {
      centers: { visible: false},
      addCenter: { visible: false},
      modifyCenter: { visible: false},
      centerDetails: { visible: false}
    }

    this.setState({
      ...hiddenElems,
      [elem]: { visible: true}
    });
  }


  render() {
    return(
      <div className="container-fluid admin">
        <div className='row'>
          <div className="col-sm-2 side">
            <div className="list-group sidebar">
              <a href="#" className="list-group-item list-group-item-action " onClick={ () => this.show('centers') }>Centers</a>
              <a  href="#" className="list-group-item list-group-item-action" onClick={ () => this.show('addCenter') }>Add Center</a>
              <a href="#" className="list-group-item list-group-item-action" >Events</a>
              <a href="#" className="list-group-item list-group-item-action">Users</a>
            </div>
          </div>
          <div className="col-sm-10 panel-column">
            { (this.state.centers.visible) && <AdminCenters show={this.show} /> }
            { (this.state.addCenter.visible) && <AddCenter show={this.show} /> }
            { (this.state.modifyCenter.visible) && <ModifyCenter /> }
            { (this.state.centerDetails.visible) && <AdminCenterDetails /> }
          </div>
        </div>
      </div>
      )
  }
}

export default AdminPanelBody;