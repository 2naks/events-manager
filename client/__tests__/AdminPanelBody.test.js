import React, { Components } from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import AdminPanelBody from '../components/AdminPanelBody.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('AdminPanelBody Component', () => {

  const initialState = {
    modifyCenter: {
      visible: 'true'
    },
    centerDetails: {
      visible: 'true'
    }
  };

  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => {
      return str;
    }
  }

  const props = {
    dispatch: () => {},
    show: () => {},
    allCenters: {
      data:[]
    }
  };

  describe('AdminPanelBody should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<AdminPanelBody {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('AdminPanelBody handle events', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<AdminPanelBody {...props} />);
      let centerButton = wrapper.find('.centers-button');
      let addCenterButton = wrapper.find('.add-center-button');
      centerButton.simulate('click');
      addCenterButton.simulate('click');
    });



  });

});
