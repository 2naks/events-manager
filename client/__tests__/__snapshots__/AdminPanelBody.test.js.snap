// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanelBody Component AdminPanelBody should render self it should render for dumb component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanelBody
    allCenters={
      Object {
        "data": Array [],
      }
    }
    dispatch={[Function]}
    show={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-sm-2 side"
        >
          <div
            className="list-group sidebar"
          >
            centerId
            <a
              className="list-group-item list-group-item-action centers-button"
              href="#"
              onClick={[Function]}
            >
              Centers
            </a>
            <a
              className="list-group-item list-group-item-action add-center-button"
              href="#"
              onClick={[Function]}
            >
              Add Center
            </a>
            <a
              className="list-group-item list-group-item-action"
              href="#"
            >
              Events
            </a>
            <a
              className="list-group-item list-group-item-action"
              href="#"
            >
              Users
            </a>
          </div>
        </div>
        <div
          className="col-sm-10 panel-column"
        >
          <Connect(AdminCenters)
            show={[Function]}
          />
        </div>
      </div>,
      "className": "container-fluid admin",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-sm-2 side"
          >
            <div
              className="list-group sidebar"
            >
              centerId
              <a
                className="list-group-item list-group-item-action centers-button"
                href="#"
                onClick={[Function]}
              >
                Centers
              </a>
              <a
                className="list-group-item list-group-item-action add-center-button"
                href="#"
                onClick={[Function]}
              >
                Add Center
              </a>
              <a
                className="list-group-item list-group-item-action"
                href="#"
              >
                Events
              </a>
              <a
                className="list-group-item list-group-item-action"
                href="#"
              >
                Users
              </a>
            </div>
          </div>,
          <div
            className="col-sm-10 panel-column"
          >
            <Connect(AdminCenters)
              show={[Function]}
            />
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="list-group sidebar"
            >
              centerId
              <a
                className="list-group-item list-group-item-action centers-button"
                href="#"
                onClick={[Function]}
              >
                Centers
              </a>
              <a
                className="list-group-item list-group-item-action add-center-button"
                href="#"
                onClick={[Function]}
              >
                Add Center
              </a>
              <a
                className="list-group-item list-group-item-action"
                href="#"
              >
                Events
              </a>
              <a
                className="list-group-item list-group-item-action"
                href="#"
              >
                Users
              </a>
            </div>,
            "className": "col-sm-2 side",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "centerId",
                <a
                  className="list-group-item list-group-item-action centers-button"
                  href="#"
                  onClick={[Function]}
                >
                  Centers
                </a>,
                <a
                  className="list-group-item list-group-item-action add-center-button"
                  href="#"
                  onClick={[Function]}
                >
                  Add Center
                </a>,
                <a
                  className="list-group-item list-group-item-action"
                  href="#"
                >
                  Events
                </a>,
                <a
                  className="list-group-item list-group-item-action"
                  href="#"
                >
                  Users
                </a>,
              ],
              "className": "list-group sidebar",
            },
            "ref": null,
            "rendered": Array [
              "centerId",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Centers",
                  "className": "list-group-item list-group-item-action centers-button",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Centers",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Center",
                  "className": "list-group-item list-group-item-action add-center-button",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add Center",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Events",
                  "className": "list-group-item list-group-item-action",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Events",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Users",
                  "className": "list-group-item list-group-item-action",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Users",
                "type": "a",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(AdminCenters)
                show={[Function]}
              />,
              false,
              false,
              false,
            ],
            "className": "col-sm-10 panel-column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "show": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-sm-2 side"
          >
            <div
              className="list-group sidebar"
            >
              centerId
              <a
                className="list-group-item list-group-item-action centers-button"
                href="#"
                onClick={[Function]}
              >
                Centers
              </a>
              <a
                className="list-group-item list-group-item-action add-center-button"
                href="#"
                onClick={[Function]}
              >
                Add Center
              </a>
              <a
                className="list-group-item list-group-item-action"
                href="#"
              >
                Events
              </a>
              <a
                className="list-group-item list-group-item-action"
                href="#"
              >
                Users
              </a>
            </div>
          </div>
          <div
            className="col-sm-10 panel-column"
          >
            <Connect(AdminCenters)
              show={[Function]}
            />
          </div>
        </div>,
        "className": "container-fluid admin",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-sm-2 side"
            >
              <div
                className="list-group sidebar"
              >
                centerId
                <a
                  className="list-group-item list-group-item-action centers-button"
                  href="#"
                  onClick={[Function]}
                >
                  Centers
                </a>
                <a
                  className="list-group-item list-group-item-action add-center-button"
                  href="#"
                  onClick={[Function]}
                >
                  Add Center
                </a>
                <a
                  className="list-group-item list-group-item-action"
                  href="#"
                >
                  Events
                </a>
                <a
                  className="list-group-item list-group-item-action"
                  href="#"
                >
                  Users
                </a>
              </div>
            </div>,
            <div
              className="col-sm-10 panel-column"
            >
              <Connect(AdminCenters)
                show={[Function]}
              />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="list-group sidebar"
              >
                centerId
                <a
                  className="list-group-item list-group-item-action centers-button"
                  href="#"
                  onClick={[Function]}
                >
                  Centers
                </a>
                <a
                  className="list-group-item list-group-item-action add-center-button"
                  href="#"
                  onClick={[Function]}
                >
                  Add Center
                </a>
                <a
                  className="list-group-item list-group-item-action"
                  href="#"
                >
                  Events
                </a>
                <a
                  className="list-group-item list-group-item-action"
                  href="#"
                >
                  Users
                </a>
              </div>,
              "className": "col-sm-2 side",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "centerId",
                  <a
                    className="list-group-item list-group-item-action centers-button"
                    href="#"
                    onClick={[Function]}
                  >
                    Centers
                  </a>,
                  <a
                    className="list-group-item list-group-item-action add-center-button"
                    href="#"
                    onClick={[Function]}
                  >
                    Add Center
                  </a>,
                  <a
                    className="list-group-item list-group-item-action"
                    href="#"
                  >
                    Events
                  </a>,
                  <a
                    className="list-group-item list-group-item-action"
                    href="#"
                  >
                    Users
                  </a>,
                ],
                "className": "list-group sidebar",
              },
              "ref": null,
              "rendered": Array [
                "centerId",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Centers",
                    "className": "list-group-item list-group-item-action centers-button",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Centers",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Center",
                    "className": "list-group-item list-group-item-action add-center-button",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Add Center",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Events",
                    "className": "list-group-item list-group-item-action",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Events",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Users",
                    "className": "list-group-item list-group-item-action",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Users",
                  "type": "a",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(AdminCenters)
                  show={[Function]}
                />,
                false,
                false,
                false,
              ],
              "className": "col-sm-10 panel-column",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
              false,
              false,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
